### ISSUE TITLE
Training stops deterministically before reaching training_steps

### ISSUE THREAD

**Issue by prachiAeromana** [aws/sagemaker-python-sdk#380]
Please fill out the form below.

### System Information
- **Framework (e.g. TensorFlow) / Algorithm (e.g. KMeans)**: Tensorflow 1.8.0
- **Framework Version**:
- **Python Version**: 3.6
- **CPU or GPU**: GPU
- **Python SDK Version**: 
- **Are you using a custom image**: No

### Describe the problem

Training ends at the same step (but depends on resolution of images provided for training) each time regardless of what training_steps is set to (when training_steps is greater than the limits below). I am training an object detection model. For images of size 2048 x 2048, it always stops at step 84. When training with images of size 1024x1024, it always stops at step 336. When training with images of size 768x768, it always stops at step 3094. I can restart training again, and run the same number of steps again. For example, for images of size 1024x1024, if I run again, it will stop at step 672. I do not receive any errors except for "Failed to run optimizer ArithmeticOptimizer, stage HoistCommonFactor. Error: Node ArithmeticOptimizer/HoistCommonFactor_Add_AddN is missing output properties at position :0 (num_outputs=0)" which it seems can be ignored, but I am not sure about that... 

Is there anything that could cause training to end in this way?

### Minimal repro / logs 
Possibly relevant output with verbose parts removed

2018-09-05 00:17:23.728002: W ./tensorflow/core/grappler/optimizers/graph_optimizer_stage.h:233] Failed to run optimizer ArithmeticOptimizer, stage HoistCommonFactor. Error: Node ArithmeticOptimizer/HoistCommonFactor_Add_AddN is missing output properties at position :0 (num_outputs=0)
2018-09-05 00:17:52,034 INFO - tensorflow - loss = 314.98633, step = 0
...
2018-09-05 00:18:54,469 INFO - tensorflow - Saving checkpoints for 26 into s3://sagemaker-tutorials1/saved_ckpts/retinanet12/model.ckpt.
...
2018-09-05 00:19:17,722 INFO - tensorflow - s3://sagemaker-tutorials1/saved_ckpts/retinanet12/model.ckpt-26 is not in all_model_checkpoint_paths. Manually adding it.
...
2018-09-05 00:19:22,011 INFO - tensorflow - Skip the current checkpoint eval due to throttle secs (600 secs).
2018-09-05 00:20:31,154 INFO - tensorflow - Saving checkpoints for 84 into s3://sagemaker-tutorials1/saved_ckpts/retinanet12/model.ckpt.
...
2018-09-05 00:20:46,048 INFO - tensorflow - s3://sagemaker-tutorials1/saved_ckpts/retinanet12/model.ckpt-84 is not in all_model_checkpoint_paths. Manually adding it.
...
2018-09-05 00:20:49,098 INFO - tensorflow - Skip the current checkpoint eval due to throttle secs (600 secs)
...
2018-09-05 00:20:50,489 INFO - tensorflow - Loss for final step: 1008.55865.
2018-09-05 00:20:50,493 INFO - tensorflow - Training has already ended. But the last eval is skipped due to eval throttle_secs. Now evaluating the final checkpoint.
...
2018-09-05 00:20:51,512 INFO - tensorflow - Calling model_fn.
2018-09-05 00:20:56,084 INFO - tensorflow - Done calling model_fn.
2018-09-05 00:20:56,104 INFO - tensorflow - Starting evaluation at 2018-09-05-00:20:56
2018-09-05 00:20:57,317 INFO - tensorflow - Graph was finalized.
2018-09-05 00:20:57.318178: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1471] Adding visible gpu devices: 0
2018-09-05 00:20:57.318215: I tensorflow/core/common_runtime/gpu/gpu_device.cc:952] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-09-05 00:20:57.318229: I tensorflow/core/common_runtime/gpu/gpu_device.cc:958]      0 
2018-09-05 00:20:57.318234: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] 0:   N 
2018-09-05 00:20:57.318408: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1084] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 14880 MB memory) -> physical GPU (device: 0, name: Tesla V100-SXM2-16GB, pci bus id: 0000:00:1e.0, compute capability: 7.0)
2018-09-05 00:20:57,318 INFO - tensorflow - Restoring parameters from s3://sagemaker-tutorials1/saved_ckpts/retinanet12/model.ckpt-84
...
2018-09-05 00:21:00,375 INFO - tensorflow - Running local_init_op.
2018-09-05 00:21:00,444 INFO - tensorflow - Done running local_init_op.

2018-09-05 00:21:01.922499: W ./tensorflow/core/grappler/optimizers/graph_optimizer_stage.h:233] Failed to run optimizer ArithmeticOptimizer, stage HoistCommonFactor. Error: Node ArithmeticOptimizer/HoistCommonFactor_Add_AddN is missing output properties at position :0 (num_outputs=0)
2018-09-05 00:21:13,489 INFO - tensorflow - Evaluation [1/1]
2018-09-05 00:21:14,997 INFO - tensorflow - Finished evaluation at 2018-09-05-00:21:14
2018-09-05 00:21:14,999 INFO - tensorflow - Saving dict for global step 84: box_loss = 2.8119822, cls_loss = 0.6933516, global_step = 84, loss = 1226.3789
...
2018-09-05 00:21:16,023 INFO - tensorflow - Saving 'checkpoint_path' summary for global step 84: s3://sagemaker-tutorials1/saved_ckpts/retinanet12/model.ckpt-84
...
2018-09-05 00:21:17.039413: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180905T0021171536106877027
2018-09-05 00:21:17,044 INFO - tensorflow - Calling model_fn.
2018-09-05 00:21:20,600 INFO - tensorflow - Done calling model_fn.
2018-09-05 00:21:20,600 INFO - tensorflow - Signatures INCLUDED in export for Eval: None
2018-09-05 00:21:20,600 INFO - tensorflow - Signatures INCLUDED in export for Classify: None
2018-09-05 00:21:20,601 INFO - tensorflow - Signatures INCLUDED in export for Regress: None
2018-09-05 00:21:20,601 INFO - tensorflow - Signatures INCLUDED in export for Predict: ['serving_default']
2018-09-05 00:21:20,601 INFO - tensorflow - Signatures INCLUDED in export for Train: None
2018-09-05 00:21:20.601338: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1471] Adding visible gpu devices: 0
2018-09-05 00:21:20.601373: I tensorflow/core/common_runtime/gpu/gpu_device.cc:952] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-09-05 00:21:20.601386: I tensorflow/core/common_runtime/gpu/gpu_device.cc:958]      0 
2018-09-05 00:21:20.601392: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] 0:   N 
2018-09-05 00:21:20.601544: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1084] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 14880 MB memory) -> physical GPU (device: 0, name: Tesla V100-SXM2-16GB, pci bus id: 0000:00:1e.0, compute capability: 7.0)
2018-09-05 00:21:20,973 INFO - tensorflow - Restoring parameters from s3://sagemaker-tutorials1/saved_ckpts/retinanet12/model.ckpt-84
...
2018-09-05 00:21:23,617 INFO - tensorflow - Assets added to graph.
2018-09-05 00:21:23,617 INFO - tensorflow - No assets to write.
2018-09-05 00:21:23.618045: I tensorflow/core/platform/s3/aws_logging.cc:54] Connection has been released. Continuing.
2018-09-05 00:21:23.627505: E tensorflow/core/platform/s3/aws_logging.cc:60] No response body. Response code: 404
...
2018-09-05 00:21:34.084794: I tensorflow/core/platform/s3/aws_logging.cc:54] Deleting file: /tmp/s3_filesystem_XXXXXX20180905T0021331536106893958
2018-09-05 00:21:34,085 INFO - tensorflow - SavedModel written to: s3://sagemaker-tutorials1/saved_ckpts/retinanet12/export/Servo/temp-1536106876/saved_model.pb
...
2018-09-05 00:21:36,552 INFO - tf_container - Downloaded saved model at /opt/ml/model/export/Servo/1536106876/saved_model.pb
